go/api/base_client.go

```diff
@@ -6429,3 +6429,113 @@ func (client *baseClient) ZLexCount(key string, rangeQuery *options.RangeByLex)
 	}
 	return handleIntResponse(result)
 }
+
+//	Blocks the connection until it pops and returns a member-score pair
+//
+// with the highest score from the first non-empty sorted set.
+//
+// See [valkey.io] for details.
+//
+// Parameters:
+//
+//	keys - An array of keys to check for elements.
+//	timeoutSecs - The maximum number of seconds to block (0 blocks indefinitely).
+//
+// Return value:
+//
+//	A `KeyWithMemberAndScore` struct containing the key from which the member was popped,
+//	the popped member, and its score. If no element could be popped and the timeout expired,
+//	returns `nil`.
+//
+// [valkey.io]: https://valkey.io/commands/bzpopmax/
+func (client *baseClient) BZPopMax(
+	keys []string,
+	timeoutSecs float64,
+) (Result[KeyWithMemberAndScore], error) {
+	args := append(keys, utils.FloatToString(timeoutSecs))
+
+	result, err := client.executeCommand(C.BZPopMax, args)
+	if err != nil {
+		return CreateNilKeyWithMemberAndScoreResult(), err
+	}
+
+	return handleKeyWithMemberAndScoreResponse(result)
+}
+
+// ZMPopWithCount behaves like ZMPop but allows specifying a maximum number of elements to pop.
+//
+// Parameters:
+//   - keys: An array of keys to check for elements.
+//   - scoreFilter: Pop criteria - either [api.MIN] or [api.MAX] to pop members with the lowest/highest scores.
+//   - count: The maximum number of elements to pop.
+//
+// Returns:
+//   - Result containing a KeyWithArrayOfMembersAndScores object or nil if no elements could be popped.
+//
+// [valkey.io]: https://valkey.io/commands/zmpop/
+func (client *baseClient) ZMPopWithCount(
```

Comment:
```suggestion
func (client *baseClient) ZMPopWithOptions(
```
And use `options.ZPopOptions` as the last argument. See `ZPopMinWithOptions` for example